### PR TITLE
SAL annotation fixes

### PR DESCRIPTION
**Pull request by meg-gupta**

### CODE REVIEW

File: lib/Backend/InterpreterThunkEmitter.cpp [hunk 1]
```diff
@@ -393,7 +393,7 @@ void InterpreterThunkEmitter::FillBuffer(
     _In_ bool asmJsThunk,
     _In_ intptr_t finalAddr,
     _In_ size_t bufferSize,
-    _Out_writes_bytes_all_(bufferSize) BYTE* buffer,
+    _Out_writes_bytes_all_(BlockSize) BYTE* buffer,
```
Comment (ThomsonTan) [inline lib/Backend/InterpreterThunkEmitter.cpp]: >buffer [](start = 44, length = 6)

Could the function use buffer out side of bufferSize?

Comment (meg-gupta) [inline lib/Backend/InterpreterThunkEmitter.cpp]: Doesn't look like. It doesn't uses BlockSize.

**Comment by meg-gupta** [comment]
@curtisman , @ThomsonTan Can you please review ?

File: lib/Backend/InterpreterThunkEmitter.cpp (hunk 1 above)
Comment (ThomsonTan) [inline lib/Backend/InterpreterThunkEmitter.cpp]: Why annotating buffer with `BlockSize` if it doesn't use `BlockSize`?

**Comment by ThomsonTan** [comment]
:shipit:

**Pull request merged**